test/jdk/java/lang/StringBuffer/ECoreIndexOf.java

```diff
@@ -0,0 +1,1370 @@
+      foo += indexOfKernel("\u01fe", "\u01ff");
+      foo += indexOfKernel("\u01fe", "\u01ff", 0);
+      foo += indexOfKernel("a", "a");
+      foo += indexOfKernel("a", "a", 0);
+      foo += indexOfKernel("a", "\u01ff");
+      foo += indexOfKernel("a", "\u01ff", 0);
+
+      foo += indexOfKernel("\u01fe", c);
+      foo += indexOfKernel("\u01fe", c, 0);
+      foo += indexOfKernel("\u01fe", c16);
+      foo += indexOfKernel("\u01fe", c16, 0);
+      foo += indexOfKernel("a", c);
+      foo += indexOfKernel("a", c, 0);
+      foo += indexOfKernel("a", c16);
+      foo += indexOfKernel("a", c16, 0);
+
+      foo += indexOfKernel(sb16, c);
+      foo += indexOfKernel(sb16, c, 0);
+      foo += indexOfKernel(sb16, c16);
+      foo += indexOfKernel(sb16, c16, 0);
+      foo += indexOfKernel(sb, c);
+      foo += indexOfKernel(sb, c, 0);
+      foo += indexOfKernel(sb, c16);
+      foo += indexOfKernel(sb, c16, 0);
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////////
+    ///////////////////////////////////////////////////////////////////////////////////
+
+    String[] decorators = {"", " (same char)"};
+    Charset[] charSets = {StandardCharsets.ISO_8859_1, StandardCharsets.UTF_16};
+    boolean[] truefalse = {true, false};
+
+    titles.put(StandardCharsets.ISO_8859_1, "L");
+    titles.put(StandardCharsets.UTF_16, "U");
+
+    for (int xxy = 0; xxy < 2; xxy++) { // Run at least twice to ensure stub called
+
+      for (int i = 0; i < 128; i++) {
+        haystack[i] = (char) i;
+      }
+
+      haystack_16[0] = '\u0000'; // (char) (23 + 256);
+      for (int i = 1; i < 128; i++) {
+        haystack_16[i] = (char) (i);
+      }
+
+      simpleTest();
+      compareIndexOfLastIndexOf();
+      compareStringStringBuffer();
+      StringIndexof();
+      StringIndexofChar();
+      StringIndexofHuge();
+
+      for (String decorator : decorators) {
+        for (Charset csHaystack : charSets) {
+          for (Charset csNeedle : charSets) {
+            System.out.println("Testing " + titles.get(csHaystack) + titles.get(csNeedle) + decorator);
+            for (boolean useOffset : truefalse) {
+              for (boolean useBuffer : truefalse) {
+                exhaustive(useOffset, useBuffer, csHaystack, csNeedle);
+              }
+            }
+          }
+        }
+
+        for (int i = 0; i < 128; i++) {
+          haystack[i] = (char) 'a';
+        }
+
+        for (int i = 0; i < 128; i++) {
+          haystack_16[i] = (char) ('a' + 256);
+        }
+      }
+    }
+
+    System.out.println(testName + " complete.");
+
+    if (failure)
+      throw new RuntimeException("One or more failures.");
+  }
+
+  private static void report(String testName, int failCount) {
+    System.err.println(testName + ": " +
+        (failCount == 0 ? "Passed" : "Failed(" + failCount + ")"));
+    if (failCount > 0)
+      failure = true;
+  }
+
+  private static String generateTestString(int min, int max) {
```

Comment:
I see you have various `Charset[] charSets` above, but this function still only generates LL. Are those separate tests? Or am I missing some concatenation somewhere that will convert the generated string string to the correct encoding?

You could had implemented my suggestion from IndexOf.generateTestString here instead, so that the tests that do call this function endup with multiple encodings; i.e. similar to what you already do in the next function.

I suppose, with addition of String/IndexOf.java that is a moot point.

Reply:
Yes, I think it's a moot point.  Thanks.